common/persistence/sql/nexus_service_store.go

```diff
@@ -105,7 +105,7 @@ func (s *sqlNexusIncomingServiceStore) ListNexusIncomingServices(
 	ctx context.Context,
 	request *p.InternalListNexusIncomingServicesRequest,
 ) (*p.InternalListNexusIncomingServicesResponse, error) {
-	lastServiceID := ""
+	lastServiceID := make([]byte, 0)
```

Comment:
Actually for some reason all 3 plugins don't like the zero value of byte arrays for keyset pagination. We do something similar for the `MinTaskQueueId` here: https://github.com/temporalio/temporal/blob/main/common/persistence/sql/task.go#L60

I did make a single variable for the empty service ID though to avoid reallocating every time.